master/internal/api/middlewares.go

```diff
@@ -0,0 +1,20 @@
+package api
+
+import (
+	"github.com/labstack/echo"
+	"github.com/labstack/echo/middleware"
+)
+
+// CORSWithTargetedOrigin builds on labstack/echo CORS by dynamically setting the origin header to
+// the request's origin.
+func CORSWithTargetedOrigin(next echo.HandlerFunc) echo.HandlerFunc {
```

Comment:
question: how is dynamic setting the cors header to the request origin functionally any different from origin = "*" always?

Reply:
browsers want it to be explicitly specified/allowed. otherwise they complain

Reply:
especially if the request is carrying credentials